core/src/main/java/com/microsoft/applicationinsights/internal/util/MapUtil.java

```diff
@@ -33,14 +36,21 @@
 public class MapUtil
 {
     public static <Value> void copy(Map<String, Value> source, Map<String, Value> target) {
+        if (source == null || target == null) {
+            return;
```

Comment:
Let's use `Preconditions.checkArgument(source != null, "source must not be null")`. Similar for `target`.